servers/services/src/main/java/org/projectnessie/services/hash/HashValidator.java

```diff
@@ -28,7 +27,7 @@ public interface HashValidator {
 
   /** Validates that a hash has been provided (absolute or relative). */
   HashValidator REQUIRED_HASH =
-      (name, parsed) -> Objects.requireNonNull(parsed, String.format("%s must be provided.", name));
+      (name, parsed) -> checkArgument(parsed != null, String.format("%s must be provided.", name));
```

Comment:
Is there a difference on the client side between throwing NPE or IAE here?

Reply:
Yes, NPE -> 500, IAE -> 400. I think the latter is more correct from the client's perspective.